### PR TITLE
SR-14221: Add tricky parsing test case for trailing closure

### PR DESCRIPTION
**Pull request by chenhongjing**
This PR adds a tricky parsing test case to ensure there is a correct interpretation of trailing closure.

Resolves [SR-14221](https://bugs.swift.org/browse/SR-14221)

### CODE REVIEW

**Comment by chenhongjing** [comment]
@swift-ci Please smoke test

**Comment by chenhongjing** [comment]
@varungandhi-apple Could you please review it? Thanks.

**Comment by chenhongjing** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
Hi, thanks for submitting the patch. Couple of points:

1. I think it would be helpful for this to be in a file with tests for trailing closures, instead of being its own file. I think you can rename `test/Parse/multiple_trailing_closures.swift` to `trailing_closures.swift` and add this check there.
2. New contributors can't trigger CI due to some trouble we've had in the past, but I can do that for you after you're done with making changes.

File: test/Parse/trailing_closure.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: %target-typecheck-verify-swift
+
+func f()->Int{ 42 }
+
+//This should be interpreted as a trailing closure, instead of being interpreted as a computed property with undesired initial value.
```
Comment (varungandhi-apple) [inline test/Parse/trailing_closure.swift:5]: Nit: Please try to keep the spacing and line widths consistent with the rest of the project. This means that comments should have a space after the `//`, and they need to be wrapped to 80 columns.

For functions, the arrow should have spaces on both sides (and similarly, there needs to be a space before an opening brace).

```
func f()->Int{ 42 } // right now
func f() -> Int { 42 } // match existing style
```

Comment (chenhongjing) [reply test/Parse/trailing_closure.swift]: Thanks for your kind instruction. I've made the changes. A small question--when will I be qualified to trigger CI?

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

File: test/Parse/trailing_closure.swift (hunk 1 above)
Comment (varungandhi-apple) [reply test/Parse/trailing_closure.swift]: There is a commit access section on https://swift.org/contributing/#contributing-code which describes the details. Usually commit access and swift-ci access are granted together.

**Comment by varungandhi-apple** [comment]
CI is passing; merging! 🚢

**Pull request merged**